internal/resources/ekscluster/resource_ekscluster.go

```diff
@@ -571,11 +581,49 @@ func resourceClusterInPlaceUpdate(ctx context.Context, d *schema.ResourceData, m
 	return dataSourceTMCEKSClusterRead(ctx, d, m)
 }
 
+func resourceClusterImporter(ctx context.Context, d *schema.ResourceData, m interface{}) ([]*schema.ResourceData, error) {
+	config := m.(authctx.TanzuContext)
+
+	id := d.Id()
+	if id == "" {
+		return nil, errors.New("ID is needed to import an TMC EKS cluster")
+	}
+
+	resp, err := config.TMCConnection.EKSClusterResourceService.EksClusterResourceServiceGetByID(id)
+	if err != nil {
+		return nil, errors.Wrapf(err, "Unable to get Tanzu Mission Control EKS cluster entry for id %s", id)
+	}
+
+	npresp, err := config.TMCConnection.EKSNodePoolResourceService.EksNodePoolResourceServiceList(resp.EksCluster.FullName)
+	if err != nil {
+		return nil, errors.Wrapf(err, "Unable to get Tanzu Mission Control EKS nodepools for cluster %s", resp.EksCluster.FullName.Name)
+	}
+
+	if err = d.Set(CredentialNameKey, resp.EksCluster.FullName.CredentialName); err != nil {
+		return nil, errors.Wrapf(err, "Failed to set credential name for the cluster %s", resp.EksCluster.FullName.Name)
+	}
+
+	if err = d.Set(RegionKey, resp.EksCluster.FullName.Region); err != nil {
+		return nil, errors.Wrapf(err, "Failed to set region for the cluster %s", resp.EksCluster.FullName.Name)
+	}
+
+	if err = d.Set(NameKey, resp.EksCluster.FullName.Name); err != nil {
+		return nil, errors.Wrapf(err, "Failed to set name for the cluster %s", resp.EksCluster.FullName.Name)
+	}
+
+	err = setResourceData(d, resp.EksCluster, npresp.Nodepools)
```

Comment:
I am not sure about the behaviour here, just wanted to understand if we need to set ID in terraform state explicitly for imported resource ? I mean this  - d.SetId(id)

Reply:
Since the ID that we are using for import matches the ID that we would set, we don't need to explicitly set the ID. Ref: https://developer.hashicorp.com/terraform/plugin/sdkv2/resources/import